client/audio/audio.cpp

```diff
@@ -78,40 +78,26 @@ const QVector<QString> *get_soundplugin_list(const struct option *poption)
   return plugin_list;
 }
 
-/**
-   Returns a static string vector of audiosets of given type available
-   on the system by searching all data directories for files matching
-   suffix.
-   The list is nullptr-terminated.
- */
-static const QVector<QString> *
-get_audio_speclist(const char *suffix, QVector<QString> **audio_list)
-{
-  if (nullptr == *audio_list) {
-    *audio_list = fileinfolist(get_data_dirs(), suffix);
-  }
-
-  return *audio_list;
-}
-
 /**
    Returns a static string vector of soundsets available on the system.
  */
 const QVector<QString> *get_soundset_list(const struct option *poption)
 {
-  static QVector<QString> *sound_list = nullptr;
-
-  return get_audio_speclist(SNDSPEC_SUFFIX, &sound_list);
+  static QVector<QString> *sound_list = new QVector<QString>;
+  sound_list->clear();
+  *sound_list = *fileinfolist(get_data_dirs(), SNDSPEC_SUFFIX);
```

Comment:
We need to `delete` the pointer returned by `fileinfolist` or we'll have a memory leak. In addition, clearing isn't needed before an assignment.
```suggestion
  auto list = fileinfolist(get_data_dirs(), SNDSPEC_SUFFIX);
  *sound_list = std::move(*list);
  delete list;
```

Reply:
I figured a `clear()` before using it did the same. I'll fix.